rustworkx-core/src/generators/hexagonal_lattice_graph.rs

```diff
@@ -317,8 +315,119 @@ mod tests {
     #[test]
     fn test_hexagonal_lattice_error() {
         let g: petgraph::graph::UnGraph<(), ()> =
-            hexagonal_lattice_graph(0, 0, || (), || (), false).unwrap();
+            hexagonal_lattice_graph(0, 0, || (), || (), false, false).unwrap();
         assert_eq!(g.node_count(), 0);
         assert_eq!(g.edge_count(), 0);
     }
+
+    #[test]
+    fn test_hexagonal_lattice_periodic_error() {
+        match hexagonal_lattice_graph::<petgraph::graph::UnGraph<(), ()>, (), _, _, ()>(
+            5,
+            3,
+            || (),
+            || (),
+            false,
+            true,
+        ) {
+            Ok(_) => panic!("Returned a non-error"),
+            Err(e) => assert_eq!(e, crate::generators::InvalidInputError),
+        }
+    }
+
+    #[test]
+    fn test_hexagonal_lattice_graph_periodic() {
+        let expected_edges = vec![
+            (0, 1),
+            (1, 2),
+            (2, 3),
+            (3, 0),
+            (4, 5),
+            (5, 6),
+            (6, 7),
+            (7, 4),
+            (0, 4),
+            (2, 6),
+            (5, 1),
+            (7, 3),
+        ];
+        let g: petgraph::graph::UnGraph<(), ()> =
+            hexagonal_lattice_graph(2, 2, || (), || (), false, true).unwrap();
+        assert_eq!(g.node_count(), 8);
+        assert_eq!(g.edge_count(), expected_edges.len());
+        assert_eq!(
+            expected_edges,
+            g.edge_references()
+                .map(|edge| (edge.source().index(), edge.target().index()))
+                .collect::<Vec<(usize, usize)>>(),
+        );
```

Comment:
I wouldn't assert the order of the edges in the generated list because that is something we reserve the right to change. You might want to compare the equality with a `HashSet` or a sorted list though.

I am glad the hard-coded edge order didn't affect your PR though.

Reply:
Thanks for the feedback -- I ended up adding two functions to do this for directed and undirected graphs.